ocs_ci/deployment/helpers/hypershift_base.py

```diff
@@ -0,0 +1,631 @@
+
+"""
+
+
+logger = logging.getLogger(__name__)
+
+
+@retry((CommandFailed, TimeoutError), tries=3, delay=5, backoff=1)
+def get_hosted_cluster_names():
+    """
+    Get HyperShift hosted cluster names
+    Returns:
+        list: the list of hosted cluster names
+    """
+    logger.info("Getting HyperShift hosted cluster names")
+    cmd = "oc get --namespace clusters hostedclusters -o custom-columns=NAME:.metadata.name --no-headers"
+    return exec_cmd(cmd, shell=True).stdout.decode("utf-8").strip().split()
+
+
+def kubeconfig_exists_decorator(func):
+    """
+    Decorator to check if the kubeconfig exists before executing the decorated method
+    :param func: func to decorate; should be used only for methods of class having 'cluster_kubeconfig' attribute !
+    :return: wrapper
+    """
+
+    def wrapper(self, *args, **kwargs):
+        if not os.path.exists(self.cluster_kubeconfig):
+            logger.error(f"no kubeconfig found for cluster {self.name}")
+            return  # Skip executing the decorated method
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+class HyperShiftBase:
+    """
+    Class to handle HyperShift hosted cluster management
+    """
+
+    def __init__(self):
+        super().__init__()
+        bin_dir_rel_path = os.path.expanduser(config.RUN["bin_dir"])
+        self.bin_dir = os.path.abspath(bin_dir_rel_path)
+        self.hcp_binary_path = os.path.join(self.bin_dir, "hcp")
+        self.hypershift_binary_path = os.path.join(self.bin_dir, "hypershift")
+        # ocp instance for running oc commands
+        self.ocp = OCP()
+        self.icsp_mirrors_path = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="icsp_mirrors-", delete=False
+        ).name
+
+    def hcp_binary_exists(self):
+        """
+        Check if hcp binary exists
+        Returns:
+            bool: True if hcp binary exists, False otherwise
+        """
+        return os.path.isfile(self.hcp_binary_path)
+
+    def install_hcp_and_hypershift_from_git(self):
+        """
+        Install hcp binary from git
+        """
+        if self.hcp_binary_exists():
+            logger.info(
+                f"hcp binary already exists {self.hcp_binary_path}, skipping download."
+            )
+            return
+
+        hcp_version = config.ENV_DATA["hcp_version"]
+
+        logger.info("Downloading hcp binary from git")
+
+        temp_dir = tempfile.mkdtemp()
+
+        exec_cmd(
+            f"git clone --single-branch --branch release-{hcp_version} "
+            f"--depth 1 {constants.HCP_REPOSITORY} {temp_dir}"
+        )
+
+        exec_cmd(f"cd {temp_dir} && make hypershift product-cli", shell=True)
+
+        exec_cmd(f"mv {temp_dir}/bin/hypershift {self.hypershift_binary_path}")
+        exec_cmd(f"mv {temp_dir}/bin/hcp {self.hcp_binary_path}")
+        hcp_version = exec_cmd("hcp --version").stdout.decode("utf-8").strip()
+        logger.info(f"hcp binary version: {hcp_version}")
```

Comment:
this is a critical point, if we fail here the job error is acceptable. Do you suggest to add a check on hcp and hypershift files have been moved into bin dir? 

Function has been tested multiple times and never give us failures

Reply:
I think that my point was, that if the download or unpacking will be unsuccessful, it will fail here (or return invalid response), so it will be more clear to firstly do the check if it was correctly downloaded and unpacked and than check/print the version... But I agree that it is more like cosmetic thing.

Reply:
addressed:

      ```
  if not (self.hcp_binary_exists() and self.hypershift_binary_exists()):
            raise Exception("Failed to download hcp binary from git")
```